allow-comma-dangle-on-multiline

eren-zeplin opened the pull request:
Allows trailing commas when the last element or property is in a different line than the closing `]` or `}` and disallows trailing commas when the last element or property is on the same line as the closing `]` or `}`.

This can be helpful for us to have a cleaner git history and can give us cleaner visibility of difference between branches.

### **Problem:**
As an example, in the screenshot below, only real change is adding a new param called `uid`. But since we don't allow trailing commas in the current config, we also have to add a comma in the previous line.
![Screenshot 2023-01-02 at 11 48 25](https://user-images.githubusercontent.com/116061420/210210030-9d61d51d-9f2d-4024-b201-9b6559ab8127.png).
With the proposed change in this PR, whenever we want to add a new item, we can add it along with a dangling comma at the end of line. So that whenever a new item is added, only change we see in Git diff will be the newly added item.

eren-zeplin committed:
Allow comma dangle on multiline

tasdemirbahadir reviewed (approved):
I see the package-lock.json is changed whereas the package.json didn't, maybe this repo is not built on local

eren-zeplin committed:
Revert changes in package-lock.json

yuqu commented:
@eren-zeplin you can update package-lock.json. lockfile v1 is no longer supported. Also there are some audit reports, you may fix them as well using a separate commit/PR.

eren-zeplin committed:
Update package-lock.json file

eren-zeplin commented:
> @eren-zeplin you can update package-lock.json. lockfile v1 is no longer supported. Also there are some audit reports, you may fix them as well using a separate commit/PR.

Updated and fixed the audit reports.
